apps/scotty/scotty.go

```diff
@@ -151,11 +170,24 @@ func loadTestMdbChannel(count int) <-chan *mdb.Mdb {
 	return result
 }
 
-func createApplicationStats(
-	appList *datastructs.ApplicationList,
+func getMyHostName(machines []mdb.Machine, myIpAddrs []string) string {
```

Comment:
This is being called every time you get an MDB update. Please consider caching the result (or perhaps the IP address or the Machine info) to avoid this search each time.